### ISSUE TITLE
MorphismFromDirectProductToExponential is broken

### ISSUE THREAD

**Issue by TKuh** [homalg-project/FinSetsForCAP#84]
The following happens with `MorphismFromDirectProductToExponential`

```gap
gap> LoadPackage("FinSetsForCAP");
gap> a_fs := FinSet( [ 1 .. 2 ] );;
gap> b_fs := FinSet( [ 3 .. 5 ] );;
gap> MorphismFromDirectProductToExponential( a_fs, b_fs );;
Error, in function PreCompose
       of category FinSets:
       morphisms not composable
 at /home/tom/.gap/pkg/CAP_project/CAP/gap/InstallAdds.gi:36 called from
CAP_INTERNAL_DISPLAY_ERROR_FOR_FUNCTION_OF_CATEGORY( record.function_name, category, pre_func_return[2] ); at /home/tom/.gap/pkg/CAP_project/CAP/gap/InstallAdds.gi:500 called from
CallFuncList( func, arg ) at /home/tom/.gap/pkg/homalg_project/ToolsForHomalg/gap/CachingObjects.gi:771 called from
PreCompose( cat, alpha, beta ) at /home/tom/.gap/pkg/CAP_project/CAP/gap/CategoryMorphisms.gi:495 called from
func( res, list[i] ) at /home/tom/Software/gap-4.11.1/lib/list.gi:2969 called from
Iterated( morphism_list, function ( alpha, beta )
      return PreCompose( cat, alpha, beta );
  end ) at /home/tom/.gap/pkg/CAP_project/CAP/gap/CategoryMorphisms.gi:495 called from
...  at *stdin*:4
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> 
```

It's derived from [here](https://github.com/homalg-project/Toposes/blob/172e8fabd9273ae083fdf0b3fd384ccadf2434e1/gap/SymmetricCartesianClosedCategoriesDerivedMethods.gi#L501-L535), but I can't spot a mistake in the derivation. :/
Also, I have by now adapted the test from CategoryOfRows to FinSets (as discussed to test Toposes) and everything works fine, except for this.

**Comment by mohamed-barakat** [comment]
The corresponding input for `SkeletalFinSets` works:
```gap
gap> a_fs := FinSet( 2 );
<An object in SkeletalFinSets>
gap> b_fs := FinSet( 3 );
<An object in SkeletalFinSets>
gap> MorphismFromDirectProductToExponential( a_fs, b_fs );
<A morphism in SkeletalFinSets>
gap> Display( last );
[ 3, [ 3, 5, 3 ], 9 ]
```

**Comment by TKuh** [comment]
This is fixed by Toposes 2022.01-04
